File: src/apps/Andriod/app/src/main/java/com/dkellycollins/triggerfinger/managers/view/impl/debug/PlayerHitboxViewManager.java
package com.dkellycollins.triggerfinger.managers.view.impl.debug;

import android.graphics.Color;

import com.dkellycollins.triggerfinger.data.entity.ICollidable;
import com.dkellycollins.triggerfinger.data.entity.IPlayer;
import com.dkellycollins.triggerfinger.managers.entity.ICollidableEntityManager;
import com.dkellycollins.triggerfinger.managers.entity.IPlayerEntityManager;

import java.util.ArrayList;
import java.util.List;

public class PlayerHitboxViewManager extends BaseHitboxViewManager {

    private final IPlayerEntityManager _playerManager;
    private final ICollidableEntityManager _collidableManager;

    public PlayerHitboxViewManager(IPlayerEntityManager playerManager, ICollidableEntityManager collidableManager) {
        _playerManager = playerManager;
        _collidableManager = collidableManager;
    }

    @Override
    protected int getColor() {
        return Color.GREEN;
    }

    @Override
    protected Iterable<ICollidable> getCollidables() {
        List<ICollidable> collidables = new ArrayList<>();

        for(IPlayer player : _playerManager.retrieve()) {
            collidables.add(_collidableManager.retrieve(player.getCollidableId()));
        }

        return collidables;
    }
}

File: src/apps/Android-Kotlin/app/src/main/kotlin/com/dkellycollins/triggerfinger/managers/view/impl/debug/PlayerHitboxViewManager.kt
package com.dkellycollins.triggerfinger.managers.view.impl.debug

import android.graphics.Color

import com.dkellycollins.triggerfinger.data.entity.ICollidable
import com.dkellycollins.triggerfinger.data.entity.IPlayer
import com.dkellycollins.triggerfinger.managers.entity.ICollidableEntityManager
import com.dkellycollins.triggerfinger.managers.entity.IPlayerEntityManager

import java.util.ArrayList

class PlayerHitboxViewManager(private val _playerManager: IPlayerEntityManager, private val _collidableManager: ICollidableEntityManager) : BaseHitboxViewManager() {

    protected override val color: Int
        get() = Color.GREEN

    protected override val collidables: Iterable<ICollidable>
        get() {
            val collidables = ArrayList<ICollidable>()

            for (player in _playerManager.retrieve()) {
                collidables.add(_collidableManager.retrieve(player.collidableId))
            }

            return collidables
        }
}
